### PR TITLE
[stable/metricbeat] add `daemonset.args` and `deployment.args`

### LINKED ISSUE #12702: Metricbeat: allow modification of docker command arguments
**Is this a request for help?**: No

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): Feature Request

**Version of Helm and Kubernetes**:

```
Client: &version.Version{SemVer:"v2.12.3", GitCommit:"eecf22f77df5f65c823aacd2dbd30ae6c65f186e", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.12.3", GitCommit:"eecf22f77df5f65c823aacd2dbd30ae6c65f186e", GitTreeState:"clean"}
```

**Which chart**:

[metricbeat](https://github.com/helm/charts/tree/master/stable/metricbeat)

**What happened**:

There is no way to override the arguments of the commands. This is possible on [filebeat](https://github.com/helm/charts/blob/master/stable/filebeat/templates/daemonset.yaml#L65-L73) and it allows a much wider area of possibilities. We use the `args` value in filebeat to load our own configuration file from a ConfigMap by mounting the volume and using the arguments `-c "filebeat.custom.yml"`.

This is not possible on Metricbeat because we can't mount on `metricbeat.yml` (mountPath has to be unique) and we can't change the configuration file loaded.

**What you expected to happen**:

Have the same feature-set as filebeat here so it's possible to change the arguments of metricbeat.

**How to reproduce it** (as minimally and precisely as possible):

**Anything else we need to know**:

If you are wondering why we would want to use our own ConfigMap instead of the `config` value, that's because we can't use template values there. We want to point Metricbeat to the Logstash service deployed on the same release, but the logstash service's name includes the release, and since we have dynamic release names, we can't hardcode the Logstash service name. We have to specify it as `{{ .Release.Name }}-logstash:5044`, which is not possible in `values.yml`.

### PR DESCRIPTION
**Pull request by mdibaiee**
#### What this PR does / why we need it:

This PR allows users to define custom `args` to be passed to the docker command for Metricbeat daemonset and deployment, [following the filebeat chart](https://github.com/helm/charts/blob/master/stable/filebeat/templates/daemonset.yaml#L65-L73).

#### Which issue this PR fixes

  - fixes #12702 

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

@at-k @mumoshu

### CODE REVIEW

File: stable/metricbeat/README.md [hunk 1]
```diff
@@ -47,20 +47,22 @@ The following table lists the configurable parameters of the metricbeat chart an
...
-| `deployment.overrideConfig`         | If overrideConfig is not empty, metricbeat chart's default config won't be used at all.            | `{}`                                                                                        |
+| `deployment.args`                   | Custom arguments for the docker command                                                                                                                                                                                                                                                     |                                                                                             |
+| `deployment.overrideConfig`         | If overrideConfig is not empty, metricbeat chart's default config won't be used at all.                                                                                                                                                                                                     | `{}`                                                                                        |
```
Comment (mdibaiee) [inline stable/metricbeat/README.md:64]: This is a change by my Markdown linter, aligning the `|` characters.

**Comment by mdibaiee** [comment]
@at-k can we get this pull-request reviewed?

**Comment by at-k** [comment]
@mdibaiee sorry for late! I applied this PR on my local env and confirmed it work correctly. 
@mumoshu  It looks good to me, how about you?

**Comment by mdibaiee** [comment]
@at-k thanks for the response and testing this. We are using this change in production system so I'm confident about it.

**Pull request merged**